### DISCUSSION TITLE
slower than mycroft

### DISCUSSION THREAD

**Discussion by geetee24** [OpenVoiceOS/ovos-buildroot#239] [Q&A] [answered]
Same Hardware Raspberry Pi 3 as my mycroft unit which answers every question in about 2 seconds. but ovos takes a good five or six seconds to answer the same question. any way to speed up the ovos response time. this is the ovos provided rpi image

**Comment by JarbasAl** [comment]
try changing STT plugin https://openvoiceos.github.io/ovos-technical-manual/stt_plugins/#list-of-stt-plugins

mycroft sent your audio to google, we use fasterwhisper, that is the main difference

you can also use [ovos-stt-plugin-chromium](https://github.com/OpenVoiceOS/ovos-stt-plugin-chromium) if you want, but by default we wont send your audio to google

**Reply by geetee24** [reply]
please comment on the relative speed of mc, fw, ch......  and the pros and cons please.  and what makes it slower please.

**Reply by geetee24** [reply]
Is there a way to turn on ovos detailed logging so when I speak a command to ovos I can see Ashley what it's doing and how long it's taking to understand and translate my command and then actually act on the command and then actually speak something to me

**Comment by goldyfruit** [comment, marked as answer]
> please comment on the relative speed of mc, fw, ch...... and the pros and cons please. and what makes it slower please.

Speed depends on many factors but mostly network related, if you got a big audio file to send to a STT then Wi-Fi better be stable (no packet loss), DNS resolution as well, plugins as well as a role in the latency.

- Chromium plugin = Hosted by Google in their datacenter on high-end hardware with high speed connectivity.
- FasterWhisper plugin = Hosted by community members _(myself included)_ in their basement on low-end hardware with ISP connectivity.

_Keep in mind that OVOS is a community project, there is no company behind as it was the case with Mycroft AI. OVOS is all provided by community members on their spare time. :+1:_ 

> Is there a way to turn on ovos detailed logging so when I speak a command to ovos I can see Ashley what it's doing and how long it's taking to understand and translate my command and then actually act on the command and then actually speak something to me

```
ovos-config set -k log_level -v DEBUG
```

Then restart the services.

**Reply by geetee24** [reply]
how do i restore the log level back to the default level?

**Reply by goldyfruit** [reply]
```
ovos-config set -k log_level -v INFO
```

**Comment by geetee24** [comment]
how do i restart services?

even a simple - what time is it  -takes like 5 seconds to answer.  Can you
comment as to why?  It takes 1 microsec to get the time so why 5 seconds to
actually speak  it?   I have high speed internet

**Comment by goldyfruit** [comment]
> how do i restart services?

Depends how you installed OVOS.

> even a simple - what time is it -takes like 5 seconds to answer. Can you comment as to why?

I can't comment on this since for me on Raspberry Pi 4 with the GUI I got descent timings (https://www.youtube.com/shorts/Qe2OzKGHLoM).

**Comment by geetee24** [comment]
ovos from ovos provided image. on rpi 3

**Comment by geetee24** [comment]
re: ovos-config set -k log_level -v DEBUG

where do i see the logs appear?

**Reply by JarbasAl** [reply]
you can also use the ovos-logs utility, see the Readme here https://github.com/OpenVoiceOS/ovos-utils

**Comment by goldyfruit** [comment]
> ovos from ovos provided image. on rpi 3

```
systemctl --user restart ovos
```

> where do i see the logs appear?

```
~/.local/state/mycroft/
```

**Comment by geetee24** [comment]
one of my skills is "turn on my tv" - MC did it in 1 second.  OVOS is
taking 5 seconds - and no speaking to me is required from ovos.   All Ovos
has to do is recognizer the phrase and make a local HTTP call.   What
could explain why ovos is taking 5 seconds to do what?  TTS cant be
involved since nothing is spoken?

**Reply by goldyfruit** [reply]
Then it might be the STT part; as mentioned above try to switch to the Chromium plugin.

**Comment by goldyfruit** [comment]
> @geetee24 I'm trying to develop a texting skill. Do you have a link to your TV skill for reference? I'm struggling to get my skill to work. 

Please open a separate discussion.

**Reply by geetee24** [reply]
i posted it here
https://github.com/orgs/OpenVoiceOS/discussions/242

**Comment by geetee24** [comment]
where do i set this?  what file and path?

By default the global language used by mycroft-core will be used

  "stt": {
    "module": "ovos-stt-plugin-chromium"
  }

**Comment by geetee24** [comment]
ovos-config show    shows that /sst/module is the default  but command
ovos-config get -k /sst/module   fails to show that value.  so how do i set
it manually

**Comment by geetee24** [comment]
added the chromium stt and now OVOS wont speak nor recognize my questions.  total silence.   how do i see what is wrong?  what log file?

how do i revert to the default stt and get nyself unbricked?

**Reply by JarbasAl** [reply]
you already opened another thread for this, i ask once again that you can keep questions separated and in a single place, thanks!

**Comment by geetee24** [comment]
Once again, nobody answered it after days of waiting, and it was already
closed

**Reply by JarbasAl** [reply]
Opened 16h ago and not closed... We are all volunteers, there isn't a staff of people just waiting to give support, you need to be patient, we already received a lot of notifications and don't need several about the same thing. Also as mentioned we want to keep questions searchable with marked answers for future users, so let's stay on topic and respectful of other people's time.

**Comment by geetee24** [comment]
Thank you
